packages/connect/src/base/entry.ts

```diff
@@ -394,10 +414,14 @@ export class EntryNodes extends EventEmitter {
       verbose(`can relay received ${new TextDecoder().decode(msg.slice())} from ${id.toB58String()}`)
       if (u8aEquals(msg.slice(), OK)) {
         done = true
-        break
       }
+      // End receive stream after first message
+      break
```

Comment:
The  `for` loop does not actually loop, because it always breaks on first iteration. So probably should just remove the loop.

Reply:
The `for await` loop does multiple things under the hood. It call the iterator - which can include several side effects. The it consumes one messages and most importantly, once leaving the `for await` loop, it _closes_ the stream. This is the reason why there is a loop and not a single call.

Reply:
Ok, I understand. But shouldn't it be made more explicit, rather than hiding some side effect logic under the hood?

Reply:
Adding some comment to it

Reply:
done